packages/lib/slots.ts

```diff
@@ -141,6 +141,26 @@ function buildSlots({
   return slots;
 }
 
+const adjustDateRanges = (dateRanges: DateRange[], frequency: number) => {
+  if (dateRanges.length === 0) return;
+
+  const baseStart = dateRanges[0].start.clone(); // Reference start time
+  const baseEnd = dateRanges[0].end.clone();
+
+  for (let i = 1; i < dateRanges.length; i++) {
+    // we skip if the date-range to adjust is outside the base date range day. - this avoids offset on other days.
```

Comment:
base date range day is the first day in date ranges. However dateranges are from the whole month. So this only handles the first day